python_modules/dagster/dagster/_core/selector/subset_selector.py

```diff
@@ -35,8 +37,9 @@
 
 MAX_NUM = sys.maxsize
 
-T = TypeVar("T")
-DependencyGraph: TypeAlias = Mapping[str, Mapping[T, AbstractSet[T]]]
+T_Hashable = TypeVar("T_Hashable", bound=Hashable)
+Direction: TypeAlias = Literal["downstream", "upstream"]
+DependencyGraph: TypeAlias = Mapping[Direction, Mapping[T_Hashable, AbstractSet[T_Hashable]]]
```

Comment:
even thought this is a type-only change, it is complex enough and affects mental model potentially and I would do it separately